### ISSUE TITLE
WebDAV crash while taking a snapshot

### ISSUE THREAD

**Issue by miquella** [kopia/kopia#1142]
Unfortunately, I am unable to reproduce this, but figured I would at least post the stack trace.

Kopia version (installed from the Kopia APT repo):
```console
$ kopia --version
0.8.4 build: 940d10dfa29e771467b5d9d45412b889053c2ed1 from: kopia/kopia
```

While attempting to take a snapshot, I got the following failure (the next snapshot ran fine, however):

```console
$ kopia snapshot create ${HOME}
Snapshotting mark@mark-l:/home/mark ...
[47.9K blob data]
[10.7K blob data]
 ! Ignored error when processing ".var/app/us.zoom.Zoom/cache/qtsingleapp-zoom-3e8": unknown or unsupported entry type
[772B blob data]
 ! Ignored error when processing ".zoom/data/com.zoom.ipc.assistantapp__res": unknown or unsupported entry type
[127B blob data]
 ! Ignored error when processing ".zoom/data/com.zoom.ipc.assistantapp__req": unknown or unsupported entry type
[35.6K blob data]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe365fc]
goroutine 2905383 [running]:
github.com/studio-b12/gowebdav.(*Client).copymove(0xc00045a680, 0x12c5bd5, 0x4, 0xc0115f83c0, 0x4e, 0xc01154ce80, 0x3a, 0x1, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/studio-b12/gowebdav@v0.0.0-20210203212356-8244b5a5f51a/requests.go:141 +0xbc
github.com/studio-b12/gowebdav.(*Client).Rename(...)
        /home/runner/go/pkg/mod/github.com/studio-b12/gowebdav@v0.0.0-20210203212356-8244b5a5f51a/client.go:302
github.com/kopia/kopia/repo/blob/webdav.(*davStorageImpl).PutBlobInPath.func1(0x1f00000, 0xf00000)
        /home/runner/work/kopia/kopia/repo/blob/webdav/webdav_storage.go:127 +0x206
github.com/kopia/kopia/internal/retry.WithExponentialBackoffNoValue.func1(0xc037a80000, 0xc008a92d30, 0x51fb99, 0xc038e80000)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:80 +0x26
github.com/kopia/kopia/internal/retry.internalRetry(0x1855dc0, 0xc00fd48180, 0x12f2acd, 0x25, 0xc0062bae20, 0x16f81a0, 0x3b9aca00, 0x773594000, 0xa, 0x3ff8000000000000, ...)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:55 +0xc2
github.com/kopia/kopia/internal/retry.WithExponentialBackoff(...)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:31
github.com/kopia/kopia/internal/retry.WithExponentialBackoffNoValue(0x1855dc0, 0xc00fd48180, 0x12f2acd, 0x25, 0xc008a92ec0, 0x16f81a0, 0x4e, 0x30)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:79 +0xd1
github.com/kopia/kopia/repo/blob/webdav.(*davStorageImpl).PutBlobInPath(0xc0003bef60, 0x1855dc0, 0xc00fd48180, 0xc011550300, 0x7, 0xc01154ce80, 0x3a, 0x18528f0, 0xc01155d140, 0xc0146aa480, ...)
        /home/runner/work/kopia/kopia/repo/blob/webdav/webdav_storage.go:121 +0x28a
github.com/kopia/kopia/repo/blob/sharded.Storage.PutBlob(0x185cf58, 0xc0003bef60, 0x0, 0x0, 0x12c5080, 0x2, 0x1ef7d50, 0x2, 0x2, 0x1855dc0, ...)
        /home/runner/work/kopia/kopia/repo/blob/sharded/sharded.go:115 +0xfb
github.com/kopia/kopia/repo/blob/retrying.retryingStorage.PutBlob.func1(0xc01154cdc0, 0x3f, 0x3f, 0x1)
        /home/runner/work/kopia/kopia/repo/blob/retrying/retrying_storage.go:51 +0x69
github.com/kopia/kopia/internal/retry.internalRetry(0x1855dc0, 0xc00fd48180, 0xc01154cdc0, 0x3f, 0xc0062bb210, 0x16f8168, 0x3b9aca00, 0x773594000, 0xa, 0x3ff8000000000000, ...)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:55 +0xc2
github.com/kopia/kopia/internal/retry.WithExponentialBackoff(...)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:31
github.com/kopia/kopia/repo/blob/retrying.retryingStorage.PutBlob(0x1862618, 0xc0003b2e60, 0x1855dc0, 0xc00fd48180, 0xc0146aa480, 0x36, 0x18528f0, 0xc01155d140, 0xc00cd1c0df, 0x13e3cfc6c7769)
        /home/runner/work/kopia/kopia/repo/blob/retrying/retrying_storage.go:50 +0x1a5
github.com/kopia/kopia/repo/blob/logging.(*loggingStorage).PutBlob(0xc00046cb70, 0x1855dc0, 0xc00fd48180, 0xc0146aa480, 0x36, 0x18528f0, 0xc01155d140, 0x1419c2d, 0xc0146a0cf0)
        /home/runner/work/kopia/kopia/repo/blob/logging/logging_storage.go:48 +0xbc
github.com/kopia/kopia/repo/content.(*WriteManager).writePackFileNotLocked(0xc000324280, 0x1855dc0, 0xc00fd48180, 0xc0146aa480, 0x36, 0xc006974000, 0x15, 0x20, 0xc01e460000, 0x100000, ...)
        /home/runner/work/kopia/kopia/repo/content/content_manager_lock_free.go:143 +0x188
github.com/kopia/kopia/repo/content.(*WriteManager).prepareAndWritePackInternal(0xc000324280, 0x1855dc0, 0xc00fd48180, 0xc013dfafc0, 0xc00a38c000, 0x12a5, 0x800000)
        /home/runner/work/kopia/kopia/repo/content/content_manager.go:447 +0x17d
github.com/kopia/kopia/repo/content.(*WriteManager).writePackAndAddToIndex(0xc000324280, 0x1855dc0, 0xc00fd48180, 0xc013dfafc0, 0xc006747c00, 0x0, 0x0)
        /home/runner/work/kopia/kopia/repo/content/content_manager.go:408 +0x85
github.com/kopia/kopia/repo/content.(*WriteManager).addToPackUnlocked(0xc000324280, 0x1855dc0, 0xc00fd48180, 0xc0276ae200, 0x20, 0xc00a38c000, 0x12a5, 0x800000, 0x0, 0x0, ...)
        /home/runner/work/kopia/kopia/repo/content/content_manager.go:282 +0x69b
github.com/kopia/kopia/repo/content.(*WriteManager).WriteContent(0xc000324280, 0x1855dc0, 0xc00fd48180, 0xc00a38c000, 0x12a5, 0x800000, 0x0, 0x0, 0x800000, 0xc008a93900, ...)
        /home/runner/work/kopia/kopia/repo/content/content_manager.go:643 +0x56b
github.com/kopia/kopia/repo/object.(*objectWriter).prepareAndWriteContentChunk(0xc007523680, 0x0, 0xc00a38c000, 0x12a5, 0x800000, 0x0, 0x0)
        /home/runner/work/kopia/kopia/repo/object/object_writer.go:196 +0x1e7
github.com/kopia/kopia/repo/object.(*objectWriter).flushBuffer(0xc007523680, 0x0, 0x0)
        /home/runner/work/kopia/kopia/repo/object/object_writer.go:157 +0x1dc
github.com/kopia/kopia/repo/object.(*objectWriter).Result(0xc007523680, 0x0, 0x0, 0x0, 0x0)
        /home/runner/work/kopia/kopia/repo/object/object_writer.go:251 +0x9c
github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).uploadFileInternal(0xc000478140, 0x1855dc0, 0xc00fd48180, 0xc008191f00, 0xc006be4960, 0x57, 0x7fa61c90e488, 0xc006f7f2c0, 0xc000566580, 0x0, ...)
        /home/runner/work/kopia/kopia/snapshot/snapshotfs/upload.go:153 +0x659
github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).processNonDirectories.func1(0x1855dc0, 0xc00fd48180, 0x1862198, 0xc006f7f2c0, 0xc006be4960, 0x57, 0x16fa660, 0xc000052800)
        /home/runner/work/kopia/kopia/snapshot/snapshotfs/upload.go:841 +0x4d6
github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).foreachEntryUnlessCanceled.func2(0xc026531078, 0xc0171467a8)
        /home/runner/work/kopia/kopia/snapshot/snapshotfs/upload.go:503 +0x1a5
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc00fd2d3b0, 0xc00fd48300)
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x66
```

**Comment by miquella** [comment]
Another stack trace today:

```console
$ kopia snapshot create /
Snapshotting root@mark-l:/ ...
[20.6K blob data]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe365fc]
goroutine 1034063 [running]:
github.com/studio-b12/gowebdav.(*Client).copymove(0xc0004ec140, 0x12c5bd5, 0x4, 0xc01a132dc0, 0x4e, 0xc0190efe80, 0x3a, 0x1, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/studio-b12/gowebdav@v0.0.0-20210203212356-8244b5a5f51a/requests.go:141 +0xbc
github.com/studio-b12/gowebdav.(*Client).Rename(...)
        /home/runner/go/pkg/mod/github.com/studio-b12/gowebdav@v0.0.0-20210203212356-8244b5a5f51a/client.go:302
github.com/kopia/kopia/repo/blob/webdav.(*davStorageImpl).PutBlobInPath.func1(0x1f00000, 0xf00000)
        /home/runner/work/kopia/kopia/repo/blob/webdav/webdav_storage.go:127 +0x206
github.com/kopia/kopia/internal/retry.WithExponentialBackoffNoValue.func1(0xc0298e6000, 0xc01125ad30, 0x51fb99, 0xc02ace6000)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:80 +0x26
github.com/kopia/kopia/internal/retry.internalRetry(0x1855dc0, 0xc0191fe240, 0x12f2acd, 0x25, 0xc007346e20, 0x16f81a0, 0x3b9aca00, 0x773594000, 0xa, 0x3ff8000000000000, ...)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:55 +0xc2
github.com/kopia/kopia/internal/retry.WithExponentialBackoff(...)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:31
github.com/kopia/kopia/internal/retry.WithExponentialBackoffNoValue(0x1855dc0, 0xc0191fe240, 0x12f2acd, 0x25, 0xc01125aec0, 0x16f81a0, 0x4e, 0x30)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:79 +0xd1
github.com/kopia/kopia/repo/blob/webdav.(*davStorageImpl).PutBlobInPath(0xc000452900, 0x1855dc0, 0xc0191fe240, 0xc019089147, 0x7, 0xc0190efe80, 0x3a, 0x18528f0, 0xc01a111fb0, 0xc00185a880, ...)
        /home/runner/work/kopia/kopia/repo/blob/webdav/webdav_storage.go:121 +0x28a
github.com/kopia/kopia/repo/blob/sharded.Storage.PutBlob(0x185cf58, 0xc000452900, 0x0, 0x0, 0x12c5080, 0x2, 0x1ef7d50, 0x2, 0x2, 0x1855dc0, ...)
        /home/runner/work/kopia/kopia/repo/blob/sharded/sharded.go:115 +0xfb
github.com/kopia/kopia/repo/blob/retrying.retryingStorage.PutBlob.func1(0xc0190efdc0, 0x3f, 0x3f, 0x1)
        /home/runner/work/kopia/kopia/repo/blob/retrying/retrying_storage.go:51 +0x69
github.com/kopia/kopia/internal/retry.internalRetry(0x1855dc0, 0xc0191fe240, 0xc0190efdc0, 0x3f, 0xc007347210, 0x16f8168, 0x3b9aca00, 0x773594000, 0xa, 0x3ff8000000000000, ...)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:55 +0xc2
github.com/kopia/kopia/internal/retry.WithExponentialBackoff(...)
        /home/runner/work/kopia/kopia/internal/retry/retry.go:31
github.com/kopia/kopia/repo/blob/retrying.retryingStorage.PutBlob(0x1862618, 0xc00040eeb0, 0x1855dc0, 0xc0191fe240, 0xc00185a880, 0x36, 0x18528f0, 0xc01a111fb0, 0xc01dd4e3cb, 0xd1e8fed80bb)
        /home/runner/work/kopia/kopia/repo/blob/retrying/retrying_storage.go:50 +0x1a5
github.com/kopia/kopia/repo/blob/logging.(*loggingStorage).PutBlob(0xc0004c7650, 0x1855dc0, 0xc0191fe240, 0xc00185a880, 0x36, 0x18528f0, 0xc01a111fb0, 0x14224ae, 0xc012c0fef0)
        /home/runner/work/kopia/kopia/repo/blob/logging/logging_storage.go:48 +0xbc
github.com/kopia/kopia/repo/content.(*WriteManager).writePackFileNotLocked(0xc000444140, 0x1855dc0, 0xc0191fe240, 0xc00185a880, 0x36, 0xc005d18300, 0x15, 0x20, 0xc00eb64000, 0x100000, ...)
        /home/runner/work/kopia/kopia/repo/content/content_manager_lock_free.go:143 +0x188
github.com/kopia/kopia/repo/content.(*WriteManager).prepareAndWritePackInternal(0xc000444140, 0x1855dc0, 0xc0191fe240, 0xc005d289c0, 0xc014dae000, 0x37908, 0x800000)
        /home/runner/work/kopia/kopia/repo/content/content_manager.go:447 +0x17d
github.com/kopia/kopia/repo/content.(*WriteManager).writePackAndAddToIndex(0xc000444140, 0x1855dc0, 0xc0191fe240, 0xc005d289c0, 0xc019252000, 0x0, 0x0)
        /home/runner/work/kopia/kopia/repo/content/content_manager.go:408 +0x85
github.com/kopia/kopia/repo/content.(*WriteManager).addToPackUnlocked(0xc000444140, 0x1855dc0, 0xc0191fe240, 0xc0209e1220, 0x20, 0xc014dae000, 0x37908, 0x800000, 0x0, 0x0, ...)
        /home/runner/work/kopia/kopia/repo/content/content_manager.go:282 +0x69b
github.com/kopia/kopia/repo/content.(*WriteManager).WriteContent(0xc000444140, 0x1855dc0, 0xc0191fe240, 0xc014dae000, 0x37908, 0x800000, 0x0, 0x0, 0x800000, 0xc01125b900, ...)
        /home/runner/work/kopia/kopia/repo/content/content_manager.go:643 +0x56b
github.com/kopia/kopia/repo/object.(*objectWriter).prepareAndWriteContentChunk(0xc01244a300, 0x0, 0xc014dae000, 0x37908, 0x800000, 0x0, 0x0)
        /home/runner/work/kopia/kopia/repo/object/object_writer.go:196 +0x1e7
github.com/kopia/kopia/repo/object.(*objectWriter).flushBuffer(0xc01244a300, 0x0, 0x0)
        /home/runner/work/kopia/kopia/repo/object/object_writer.go:157 +0x1dc
github.com/kopia/kopia/repo/object.(*objectWriter).Result(0xc01244a300, 0x0, 0x0, 0x0, 0x0)
        /home/runner/work/kopia/kopia/repo/object/object_writer.go:251 +0x9c
github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).uploadFileInternal(0xc000366280, 0x1855dc0, 0xc0191fe240, 0xc004e65560, 0xc0124386c0, 0x34, 0x7feb8b4bad08, 0xc005d9d720, 0xc00082e160, 0x0, ...)
        /home/runner/work/kopia/kopia/snapshot/snapshotfs/upload.go:153 +0x659
github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).processNonDirectories.func1(0x1855dc0, 0xc0191fe240, 0x1862198, 0xc005d9d720, 0xc0124386c0, 0x34, 0x0, 0x0)
        /home/runner/work/kopia/kopia/snapshot/snapshotfs/upload.go:841 +0x4d6
github.com/kopia/kopia/snapshot/snapshotfs.(*Uploader).foreachEntryUnlessCanceled.func2(0x441136, 0x16fa560)
        /home/runner/work/kopia/kopia/snapshot/snapshotfs/upload.go:503 +0x1a5
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0191c0d50, 0xc0191fe580)
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x66
```

**Comment by miquella** [comment]
I've opened studio-b12/gowebdav#44 to attempt to address this. Unfortunately, I'm not sure why the request is erroring…

**Comment by miquella** [comment]
This should be resolved now that #1179 has merged.
